app.js

```diff
@@ -16,38 +16,61 @@
 
 'use strict';
 
+const WatsonConversationSetup = require('./lib/watson-conversation-setup');
+const DEFAULT_NAME = 'watson-conversation-slots-intro';
+const fs = require('fs'); // file system for loading JSON
+const vcapServices = require('vcap_services');
+const conversationCredentials = vcapServices.getCredentials('conversation');
+const watson = require('watson-developer-cloud'); // watson sdk
+
 var express = require('express'); // app server
 var bodyParser = require('body-parser'); // parser for post requests
 var Conversation = require('watson-developer-cloud/conversation/v1'); // watson sdk
 
 var app = express();
 
+require('cf-deployment-tracker-client').track();
+
 // Bootstrap application settings
 app.use(express.static('./public')); // load UI from public folder
 app.use(bodyParser.json());
 
+let workspaceID; // workspaceID will be set when the workspace is created or validated.
+
 // Create the service wrapper
-var conversation = new Conversation({
-  // If unspecified here, the CONVERSATION_USERNAME and CONVERSATION_PASSWORD env properties will be checked
-  // After that, the SDK will fall back to the bluemix-provided VCAP_SERVICES environment property
-  // username: '<username>',
-  // password: '<password>',
-  // url: 'https://gateway.watsonplatform.net/conversation/api',
-  version_date: Conversation.VERSION_DATE_2017_04_21
+const conversation = watson.conversation({
+  url: 'https://gateway.watsonplatform.net/conversation/api',
+  username: conversationCredentials.username,
```

Comment:
FYI - username/pwd should not be required here if you use the correct config key name. It will retrieve them from your .env or from VCAP_SERVICES. You should test that out.